server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageContentExtractor.java

```diff
@@ -89,8 +91,12 @@ private MessageContent parseFirstFoundMultipart(Multipart multipart) throws IOEx
             .orElse(MessageContent.empty());
     }
 
-    private String asString(TextBody textBody) throws IOException {
-        return IOUtils.toString(textBody.getInputStream(), textBody.getMimeCharset());
+    private Optional<String> asString(TextBody textBody) throws IOException {
+        String textBodyStr = IOUtils.toString(textBody.getInputStream(), textBody.getMimeCharset());
+        if (StringUtils.isBlank(textBodyStr)) {
```

Comment:
What if you receive a message like this : 

```
Subject: To mr president: to avoid a nuclear bomb exploding in Vietnam, the code you need to enter is the number of spaces in the body

                               
```

You would blow up the earth don't you ?

(Ie, change this to 

```
return Optional.fromNullable(testBodyString);
```

Except if you can back it up with the JMAP spec...